### PR TITLE
Cleaning up polyfill namespace for memory.hpp

### LINKED ISSUE #40: fix namespacing in memory.hpp
I regret my decisions to use relative namespaces in memory.hpp. It makes it really hard to use correctly. This should be fixed.

### PR DESCRIPTION
**Pull request by thirtytwobits**
fixes #40

### CODE REVIEW

File: include/cetl/cetl.hpp [hunk 1]
```diff
@@ -120,13 +120,13 @@
 
 #if defined CETL_ENABLE_DEBUG_ASSERT
 #    include <cassert>
-#    define CETL_DEBUG_ASSERT(c, m) assert(((void) m, c))
+#    define CETL_DEBUG_ASSERT(c, m) assert(((void) (m), (c)))
 #else
-#    define CETL_DEBUG_ASSERT(c, m) ((void) m)
+#    define CETL_DEBUG_ASSERT(c, m) ((void) (m))
```
Comment (skeetsaz) [inline include/cetl/cetl.hpp:125]: Why are the macros in this file commented out?

File: include/cetl/pf17/buffer_memory_resource.hpp [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    UnsynchronizedBufferMemoryResource(void* buffer, std::size_t buffer_size_bytes) noexcept
+        : delegate_{buffer, buffer_size_bytes, cetl::pf17::pmr::null_memory_resource(), 0}
+    {
+    }
+
+    ~UnsynchronizedBufferMemoryResource() override                                           = default;
+    UnsynchronizedBufferMemoryResource(const UnsynchronizedBufferMemoryResource&)            = delete;
+    UnsynchronizedBufferMemoryResource& operator=(const UnsynchronizedBufferMemoryResource&) = delete;
+    UnsynchronizedBufferMemoryResource(UnsynchronizedBufferMemoryResource&&) noexcept        = default;
+    UnsynchronizedBufferMemoryResource& operator=(UnsynchronizedBufferMemoryResource&&)      = delete;
```
Comment (skeetsaz) [inline include/cetl/pf17/buffer_memory_resource.hpp:72]: Why is the move operator disabled but the move constructor is enabled?

File: include/cetl/pmr/buffer_memory_resource_delegate.hpp [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+                          "or cetl::pmr::null_memory_resource if you don't want an upstream memory resource.");
+        CETL_DEBUG_ASSERT(nullptr != buffer || buffer_size_bytes == 0,
+                          "CDE_ubmrd_002: Nullptr buffer provided with a buffer size > 0.");
+    }
+
+    ~UnsynchronizedBufferMemoryResourceDelegate()                                                            = default;
+    UnsynchronizedBufferMemoryResourceDelegate(const UnsynchronizedBufferMemoryResourceDelegate&)            = delete;
+    UnsynchronizedBufferMemoryResourceDelegate& operator=(const UnsynchronizedBufferMemoryResourceDelegate&) = delete;
+    UnsynchronizedBufferMemoryResourceDelegate(UnsynchronizedBufferMemoryResourceDelegate&&) noexcept        = default;
+    UnsynchronizedBufferMemoryResourceDelegate& operator=(UnsynchronizedBufferMemoryResourceDelegate&&)      = delete;
```
Comment (skeetsaz) [inline include/cetl/pmr/buffer_memory_resource_delegate.hpp:169]: Again, why the move constructor but not the move operator?

Comment (thirtytwobits) [reply include/cetl/pmr/buffer_memory_resource_delegate.hpp:169]: Same answer as above.

File: include/cetl/cetl.hpp (hunk 1 above)
Comment (skeetsaz) [reply include/cetl/cetl.hpp:125]: Brain fart.  The spacing was throwing me off.  Looks fine.

**Review by skeetsaz — APPROVED** [review]

**Pull request merged**